pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -237,8 +237,11 @@ func SetDefaults_Worker(obj *Worker) {
 	if obj.MaxUnavailable == nil {
 		obj.MaxUnavailable = &DefaultWorkerMaxUnavailable
 	}
-	if obj.AllowSystemComponents == nil {
-		obj.AllowSystemComponents = &DefaultWorkerAllowSystemComponents
+	if obj.SystemComponents == nil {
+		obj.SystemComponents = &WorkerSystemComponents{}
+	}
+	if obj.SystemComponents.Allow == nil {
+		obj.SystemComponents.Allow = &DefaultWorkerSystemComponentsAllow
```

Comment:
Hmm, I'm wondering if we should actually default this value in the Shoot object or rather leave it empty if unset and use the `DefaultWorkerSystemComponentsAllow` constant in the shoot controller as a "behavioural" default (set the node label to `true` if the flag in the ShootSpec is not set).
But I am not sure, if this would still be conformant with k8s API conventions.

Reply:
I'm not sure. 

I'd like to not have magic where defaulting is happening but not visible on the Shoot CR.
If you prefer it the other way around I can change the behavior

Reply:
The magic can happen with a wrapping function, something like https://github.com/gardener/gardener/blob/7afb12735b14f49144b3a0558eca2b1ceeaba6b0/pkg/apis/core/v1beta1/helper/helper.go#L592-L595

for example:
```go
func SystemComponentsAllowed(worker *gardencorev1beta1.Worker) bool {
	return worker.SystemComponents == nil || worker.SystemComponents.Allow == nil || *worker.SystemComponents.Allow
}
```

Reply:
I've changed the behavior in https://github.com/gardener/gardener/pull/2480/commits/8c17a3de043c1e988f916f4ef0e6b5e4993228af.
Please take a look. If it looks good, I'll squash everything together and submit this for a final review

Reply:
@tim-ebert why do you suggest this? I don't think we have been consistent here in the past, but generally it's better to be explicit if we already have a dedicated field in the API. Yes, we haven't streamlined this yet for other fields, but for new fields we tried to default it, or? What do other folks think? @timuthy @mvladev @ialidzhikov @danielfoehrKn

Reply:
I've readded the defaulting with https://github.com/gardener/gardener/pull/2480/commits/c159d9da2f77cbaf1a321b71eeeb546a5f76730d but left the helper functions in as I made use of them in the botanist